File: LeetCode/Medium/Single Number II/Java/singleNumber.java
class Solution {
    public int singleNumber(int[] nums) {
        int lone = 0;
        for (int s=0; s<32; s++) {
            int bitSum = 0;
            for (int num : nums) bitSum += (num >> s) & 1;
            int lonerBit = bitSum%3;
            lone |= (lonerBit << s);
        }
        return lone;
    }
}

File: LeetCode/Medium/Single Number II/C/single_number.c
int singleNumber(int* nums, int numsSize){
    long lone = 0;
    for (int i=0; i<32; i++) {
        int bit_sum = 0;
        for (int j=0; j<numsSize; j++) bit_sum += (nums[j] >> i) & 1;
        long loner_bit = bit_sum%3;
        lone |= (loner_bit<<i);
    }
    return lone;
}
